client/components/fields/editor/EventRelatedPlannings/EmbeddedCoverageForm.tsx

```diff
@@ -187,11 +204,12 @@ export class EmbeddedCoverageFormComponent extends React.PureComponent<IProps> {
                         <List.Row>
                             <Row>
                                 <Select
+                                    key={this.state.selectedDeskId}
```

Comment:
Why are you using a key here? value changes and component doesn't respect the prop? It should be fixed inside the select component then.

Reply:
My bad. If language is set once, we check for that and only if it's not we set a new language value. So the value does update (with the same value) and that's by design. `Select` is working properly.